libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -266,22 +266,26 @@ pair<string, map<ContractDefinition const*, string>> IRGenerator::evaluateConstr
 
 	for (ASTPointer<InheritanceSpecifier> const& base: _contract.baseContracts())
 		if (FunctionDefinition const* baseConstructor = dynamic_cast<ContractDefinition const*>(
-					base->name().annotation().referencedDeclaration
+				base->name().annotation().referencedDeclaration
 			)->constructor(); baseConstructor && base->arguments())
 			baseConstructorArguments.emplace_back(
-					dynamic_cast<ContractDefinition const*>(baseConstructor->scope()),
-					base->arguments()
+				dynamic_cast<ContractDefinition const*>(baseConstructor->scope()),
+				base->arguments()
 			);
 
 	if (FunctionDefinition const* constructor = _contract.constructor())
 		for (auto const& modifier: constructor->modifiers())
-			if (FunctionDefinition const* baseConstructor = dynamic_cast<ContractDefinition const*>(
+			if (ContractDefinition const* baseContract = dynamic_cast<ContractDefinition const*>(
 				modifier->name()->annotation().referencedDeclaration
-			)->constructor(); baseConstructor && modifier->arguments())
-				baseConstructorArguments.emplace_back(
-					dynamic_cast<ContractDefinition const*>(baseConstructor->scope()),
-					modifier->arguments()
-				);
+			))
+				if (
+					FunctionDefinition const* baseConstructor = baseContract->constructor();
+					baseConstructor && modifier->arguments()
+				)
+					baseConstructorArguments.emplace_back(
+						dynamic_cast<ContractDefinition const*>(baseConstructor->scope()),
+						modifier->arguments()
+					);
```

Comment:
Maybe should be extracted as a fix for constructor PR previously merged

Reply:
You mean #8591 ?

Reply:
Yes, this part here is a fix for that PR

Reply:
Moved to another PR, will leave it here as well until other PR is merged. Is needed for CI to successfully completes all jobs